### PR TITLE
Cache doc and code improvement

### LINKED ISSUE #555: Documenting cache package

### PR DESCRIPTION
**Pull request by mantzas**
## Which problem is this PR solving?

Closes #555 and improves the API of the cache abstractions. This is a BREAKING CHANGE.

## Short description of the changes

- Introduced context to the cache methods
- Improved Go documentation

### CODE REVIEW

File: docs/other/Caching.md [hunk 1]
```diff
@@ -1,33 +0,0 @@
-# Caching
```
Comment (xenosdio) [inline docs/other/Caching.md:1]: Why are we removing the guide here?

**Comment by xenosdio** [comment]
Since there is a breaking change, even that mundane, should we update the `BREAKING.md` file?

File: docs/other/Caching.md (hunk 1 above)
Comment (mantzas) [reply docs/other/Caching.md:1]: We made the decision to remove all markdown docs in favour of Go docs.

File: BREAKING.md [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Breaking Changes Migration Guide
 
-## v.0.74.0
+## v0.75.0
+
+The `cache` package has introduced the `context.Context` as the first argument in each method and removed it for the constructors.
```
Comment (xenosdio) [inline BREAKING.md]:
```suggestion
The `cache` package has introduced the `context.Context` as the first argument in each method and removed it from the constructors.
```

**Review by xenosdio — DISMISSED** [review]

**Review by xenosdio — APPROVED** [review]

**Pull request merged**